test/batches_SUITE.erl

```diff
@@ -303,3 +306,93 @@ handle_call(N, _From, Acc) ->
 
 handle_cast(_Msg, Acc) ->
     {noreply, Acc}.
+
+%% Retry test
+
+retry_request(_) ->
+    Opts = (retry_aggregator_opts())#{pool_id => retry_request},
+    {ok, Pid} = gen_server:start_link(mongoose_aggregator_worker, Opts, []),
+    gen_server:cast(Pid, {task, key, 1}),
+    receive_task_called(0),
+    receive_task_called(1),
+    gen_server:cast(Pid, {task, key, 1}),
+    receive_task_called(0),
+    receive_task_called(1),
+    ensure_no_tasks_to_receive().
+
+retry_aggregator_opts() ->
+    #{host_type => host_type(),
+      request_callback => fun do_retry_request/2,
+      aggregate_callback => fun aggregate_sum/3,
+      verify_callback => fun validate_ok/3,
+      flush_extra => #{host_type => host_type(), origin_pid => self()}}.
+
+validate_ok(ok, _, _) ->
+    ok;
+validate_ok({error, Reason}, _, _) ->
+    {error, Reason}.
+
+%% Fails first task
+do_retry_request(Task, #{origin_pid := Pid, retry_number := Retry}) ->
```

Comment:
Variable `Task` is unused. This applies to `do_cancel_request` and `do_request_but_ignore_other_messages` as well.